src/js/controllers/imageCtrl.js

```diff
@@ -0,0 +1,35 @@
+(function (angular) {
+    angular
+        .module('flickrPOC')
+        .controller('imagesCtrl', imagesCtrl);
+
+    imagesCtrl.$inject = ['images'];
+
+    function imagesCtrl(images) {
+        /* jshint validthis: true */
+        var vm = this;
+
+        vm.getImages = function () {
+            images.loadPics()
+                .then(function (res) {
+                    //console.log('this is not from test '+ res);
+                    vm.images = res;
+                    return vm.images;
+                })
+                .catch(
+                    function(err){
+                        console.error('Error loading images ', err.status, err.data);
+                    })
+                .finally(function () {
+                    console.log("finally finished");
+                });
+
+        };
+        function activate(){
+            vm.getImages();
```

Comment:
@cxoltero 
So what I'd like you to do is move the `images.loadImages()` to the `activate()` function, allowing the `vm.getImages()` function to be deleted. This way, the controller will load, then call `activate()`, which triggers the image retrieval. 

In your tests, instead of calling `vm.getImages()` to get the images, you'll just have to instantiate the controller for each test, which you already do on line#50 (`vm = $controller('imagesCtrl');') of the test file. The reason you'll have to do that for each test is to ensure that your values are created, and not having to depend on previous tests.